Mathlib/Topology/Algebra/Algebra.lean

```diff
@@ -74,16 +78,471 @@ theorem algebraMapCLM_toLinearMap : (algebraMapCLM R A).toLinearMap = Algebra.li
   rfl
 #align algebra_map_clm_to_linear_map algebraMapCLM_toLinearMap
 
+/-- If `R` is a discrete topological ring, then any topological ring `S` which is an `R`-algebra
+  is also a topological `R`-algebra. -/
+instance DiscreteTopology.continuousSMul [TopologicalSemiring A] [DiscreteTopology R] :
+    ContinuousSMul R A := continuousSMul_of_algebraMap _ _ continuous_of_discreteTopology
+
 end TopologicalAlgebra
 
-section TopologicalAlgebra
+section TopologicalAlgebra --
+section
 
 variable {R : Type*} [CommSemiring R]
 variable {A : Type u} [TopologicalSpace A]
 variable [Semiring A] [Algebra R A]
 
 #align subalgebra.has_continuous_smul SMulMemClass.continuousSMul
 
+end
+section
+
+variable (R : Type*) [CommSemiring R] [TopologicalSpace R] [TopologicalSemiring R]
+  (A : Type*) [Semiring A] [TopologicalSpace A] [TopologicalSemiring A]
+
+/-- Continuous algebra homomorphisms between algebras. We only put the type classes that are
+  necessary for the definition, although in applications `M` and `B` will be topological algebras
+  over the topological ring `R`. -/
+structure ContinuousAlgHom (R : Type*) [CommSemiring R] (A : Type*) [Semiring A]
+    [TopologicalSpace A] (B : Type*) [Semiring B] [TopologicalSpace B] [Algebra R A] [Algebra R B]
+    extends A →ₐ[R] B where
+  cont : Continuous toFun := by continuity
+
+/-- `ContinuousAlgHomClass F R A B` asserts `F` is a type of bundled continuous `R`-agebra
+  homomorphisms `A → B`. -/
+class ContinuousAlgHomClass (F : Type*) (R : outParam (Type*)) [CommSemiring R]
```

Comment:
I think we don't actually want this class. Instead, I believe we just want to use the separate `ContinuousMapClass F A B` and `AlgHomClass F R A B`. This isn't a problem because these are `Prop`-valued classes now that `FunLike` has been unbundled from the hom-class hierarchy. If you disagree, please continue the discussion here: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/ContinuousMonoidHomClass.20etc/near/451600499

Reply:
Done.